common/app/common/configuration.scala

```diff
@@ -491,12 +492,19 @@ class GuardianConfiguration extends GuLogging {
       configuration.getStringProperty("commercial.s3.root") getOrElse s"${environment.stage.toUpperCase}/commercial"
     }
 
+    lazy val shouldIncludeLineItemsJobs = LineItemsJobs.isSwitchedOn
+
     private lazy val dfpRoot = s"$commercialRoot/dfp"
+    private lazy val gamRoot = s"$commercialRoot/gam"
     lazy val dfpPageSkinnedAdUnitsKey = s"$dfpRoot/pageskinned-adunits-v9.json"
     lazy val dfpLiveBlogTopSponsorshipDataKey = s"$dfpRoot/liveblog-top-sponsorships-v3.json"
     lazy val dfpSurveySponsorshipDataKey = s"$dfpRoot/survey-sponsorships.json"
-    lazy val dfpNonRefreshableLineItemIdsKey = s"$dfpRoot/non-refreshable-lineitem-ids-v1.json"
-    lazy val dfpLineItemsKey = s"$dfpRoot/lineitems-v7.json"
+    lazy val dfpNonRefreshableLineItemIdsKey =
+      if (shouldIncludeLineItemsJobs) s"$gamRoot/non-refreshable-line-items.json"
+      else s"$dfpRoot/non-refreshable-lineitem-ids-v1.json"
+    lazy val dfpLineItemsKey =
```

Comment:
I might be wrong, but I think a lazy val will be lazily evaluated and then fixed, so it wouldn't update with the switch changing state.

Did the switch work after deploying?

Reply:
I checked the switch when testing in CODE and I can see in the config it was true and I removed the two objects from `dfp/` that frontend reads currently while the switch is on and we had no problem. Do you think how long we might need to test it for in CODE with the objects removed (generated by Frontend) and the switch in on?

Reply:
I think maybe with the switch on and verifying the files aren't created, then turn it off and make sure the files reappear.

Reply:
That's what I exactly have done when I tested in CODE.